crates/executor/src/lib.rs

```diff
@@ -7,9 +7,30 @@
 
 //! Reth executor executes transaction in block of data.
 
+use async_trait::async_trait;
+use execution_result::ExecutionResult;
+use reth_interfaces::executor::Error;
+use reth_primitives::{Address, Block};
+
 pub mod config;
 pub mod eth_dao_fork;
+
+/// Execution result types
+pub mod execution_result;
 /// Executor
 pub mod executor;
 /// Wrapper around revm database and types
 pub mod revm_wrap;
+
+/// An executor capable of executing a block.
+#[async_trait]
+pub trait BlockExecutor {
```

Comment:
This should be put to `reth-interfaces`.

Reply:
Agree! But I move it to `reth-executor` because it needs to return `ExecutionResult`. 

My initial idea is implementing `fn take_result(&mut self) -> Option<ExecutionResult>` on `Executor`. w/ 9e50f1603618c87bc7096394998961dd448af81f
I revised it by moving `BlockExecutor` to `reth-executor` because I thought it's not clear interface.

I'll think more how to put `BlockExecutor` to `reth-interfaces`. Thanks!

Reply:
One possible solution is also putting `execution_result.rs`, that is extracted file from `executor.rs`, to `reth-interfaces`. Is it acceptable? (Even though before getting feedback, I'll try it in order to it's really possible.)

Reply:
I tried it using Generic w/ 45e0c1324845232c770dd4aca2d9c69873fed955. Is it acceptable?

Reply:
Yeah, let us leave it as generic, with the new revm this would become cleaner, and `reth-db` dependency should be moved to `reth-provider` when that refactor happens.